### PR TITLE
move server report ext to extensions

### PR DESCRIPTION
**Pull request by alanrenmsft**
move server reports extension to extension folder so that it is being treated just like other extensions. aside from moving the file, I also updated the tsconfig.json and package.json to use the latest tools. tested with an ad-hoc build and everything look fine.

@dzsquared we currently have an issue with the server report extension, once we open the dashboard tab for server report, the entire app is pretty much unusable.  #13652 , I have fixed it in this PR too, see below comments for more context.

### CODE REVIEW

**Comment by alanrenmsft** [comment]
my test build: https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=90930&view=results, the vsix is produced together with other extensions now.

**Comment by alanrenmsft** [comment]
@dzsquared I am looking into the perf issue now.

**Comment by alanrenmsft** [comment]
@dzsquared turns out this is caused by a performance of chart.js, for one of our test server, the backup size trend chart is returning 2000+ data points, chartjs is not able to handle it, I think in general we should control the number of rows for charting, I will continue investigating how to improve the performance of the chart, in the meanwhile, for this particular chart, I think we should limit it to the most recent 100 backups or something like that. what do you think?

**Comment by alanrenmsft** [comment]
the rest of queries in this extension either have top x in the query or the query won't return large set of data, so after this commit: https://github.com/microsoft/azuredatastudio/pull/14087/commits/bcb850f32ea1defcc4179d59bad5b0b0688bf80d we should be good with this extension. @dzsquared

File: build/lib/extensions.js [hunk 1]
```diff
@@ -226,6 +226,7 @@ const externalExtensions = [
     'sql-assessment',
     'sql-database-projects',
     'sql-migration',
+    'server-report'
```
Comment (Charles-Gagnon) [inline build/lib/extensions.js:229]: Would you mind putting this above the sql- ones to keep it in alphabetical order?

File: build/lib/extensions.ts [hunk 2]
```diff
@@ -260,6 +260,7 @@ const externalExtensions = [
 	'sql-assessment',
 	'sql-database-projects',
 	'sql-migration',
+	'server-report'
```
Comment (Charles-Gagnon) [inline build/lib/extensions.ts:263]: Same here - keep alphabetical

File: extensions/server-report/.vscodeignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+out/**/*.map
```
Comment (Charles-Gagnon) [inline extensions/server-report/.vscodeignore:1]: You already have a rule below to ignore all *.map files

File: extensions/server-report/package.json [hunk 4]
```diff
@@ -0,0 +1,395 @@
+{
+	"name": "server-report",
+	"displayName": "Server Reports",
```
Comment (Charles-Gagnon) [inline extensions/server-report/package.json:3]: Would be nice if you went through and localized this once you get the initial checkin done

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**